src/chunk_item.rs

```diff
@@ -40,18 +40,40 @@ impl Basic {
         let path: String = byte_interface.getattr("path")?.extract()?;
 
         let chunk_shape = chunk_spec.getattr("shape")?.extract()?;
-        let dtype: String = chunk_spec
+        let mut dtype: String = chunk_spec
             .getattr("dtype")?
             .call_method0("__str__")?
             .extract()?;
-        let fill_value = chunk_spec
-            .getattr("fill_value")?
-            .call_method0("tobytes")?
-            .extract()?;
+        let fill_value: Bound<'_, PyAny> = chunk_spec.getattr("fill_value")?;
+        let fill_value_bytes: Vec<u8>;
+        if let Ok(fill_value_downcast) = fill_value.downcast::<PyBytes>() {
+            fill_value_bytes = fill_value_downcast.as_bytes().to_vec();
+        } else if fill_value.hasattr("tobytes")? {
+            fill_value_bytes = fill_value.call_method0("tobytes")?.extract()?;
+        } else if let Ok(fill_value_downcast) = fill_value.downcast::<PyInt>() {
+            let fill_value_usize: usize = fill_value_downcast.extract()?;
+            if fill_value_usize == (0 as usize) && dtype == "object" {
```

Comment:
Turns out this varies between zarr-python versions, but it is at least consistent with `zarr-python` 3, See https://github.com/zarr-developers/zarr-python/issues/2792#issuecomment-2637954231

Reply:
Sorry meant to reply to this earlier - this condition is still kosher, though, right? I mean, the behavior is correct, that this is implicitly the v2 case that we have been discussing elsewhere and this sort of thing is not allowed in v3?

Just want to make sure I'm clear before merging since I feat I may be a bit lost in the sauce.

Reply:
I think the "string" / 0 fill value workaround was unreachable before. I've adjusted it now to match `zarr-python` 2.x.x behaviour given the discussion in https://github.com/zarr-developers/zarr-python/issues/2792

Reply:
Oh the workaround can't be called anyway since we reject variable-sized data types from the pipeline. It'd be handy if `zarr-python` resolves https://github.com/zarr-developers/zarr-python/issues/2792 so that we get a "0" instead of a 0 and no workaround is needed on our side.

Reply:
It all makes sense now. https://github.com/zarr-developers/zarr-python/issues/2792#issuecomment-2644362122. Reverted back to the `0` -> `""` fill value workaround.